addons/website_form/models/models.py

```diff
@@ -70,7 +70,7 @@ def get_authorized_fields(self, model_name):
         for field in list(fields_get):
             if 'domain' in fields_get[field] and isinstance(fields_get[field]['domain'], str):
                 del fields_get[field]['domain']
-            if fields_get[field]['readonly'] or field in MAGIC_FIELDS:
+            if 'readonly' in fields_get[field] and fields_get[field]['readonly'] or field in MAGIC_FIELDS:
```

Comment:
It seems we don't need to know if key is there, we just use the value, so you could do `fields_get[field].get('readonly')`